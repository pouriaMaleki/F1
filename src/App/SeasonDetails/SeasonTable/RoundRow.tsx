import React from "react";
import css from "./RoundRow.module.sass";
import classNames from "classnames";
import { IRound } from "../IRound";

const RoundRow = (props: IRound) => {
  return (
    <div
      className={classNames({
        [css.root]: true,
        [css.champion]: props.seasonChampion
      })}
    >
      <div className={classNames(css.data, css.roundNo)}>{props.round}</div>
      <div className={css.data}>{props.raceName}</div>
      <div className={css.data}>{props.driver}</div>
      <div className={css.data}>{props.team}</div>
      <div className={css.data}>{props.laps} Laps</div>
      <div className={css.data}>{props.grid} Grid</div>
      <div className={css.data}>{props.time}</div>
    </div>
  );
};

export { RoundRow };
