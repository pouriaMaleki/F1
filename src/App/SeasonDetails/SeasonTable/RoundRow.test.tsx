import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import { RoundRow } from "./RoundRow";

describe("Render RoundRow", () => {
  // snapshot test
  it("render correctly RoundRow component", () => {
    const props = {
      round: 1,
      raceName: "Austali",
      driver: "Lwoid",
      team: "McDAS",
      laps: "12",
      grid: "2",
      time: "1.24.32.433",
      seasonChampion: true
    };
    const RoundRowComponent = renderer.create(<RoundRow {...props} />).toJSON();
    expect(RoundRowComponent).toMatchSnapshot();
  });

  // test props
  it("render titles and highlight seasonChampion regarding props", () => {
    const props = {
        round: "1",
        raceName: "Austali",
        driver: "Lwoid",
        team: "McDAS",
        laps: "12",
        grid: "2",
        time: "1.24.32.433",
        seasonChampion: true
      },
      RoundRowComponent = mount(<RoundRow {...props} />),
      dataEls = RoundRowComponent.find(".data");

    expect(dataEls.at(0).text()).toEqual(props.round);
    expect(dataEls.at(1).text()).toEqual(props.raceName);
    expect(dataEls.at(2).text()).toEqual(props.driver);
    expect(dataEls.at(3).text()).toEqual(props.team);
    expect(dataEls.at(4).text()).toEqual(props.laps + " Laps");
    expect(dataEls.at(5).text()).toEqual(props.grid + " Grid");
    expect(dataEls.at(6).text()).toEqual(props.time);
    expect(RoundRowComponent.find(".root").hasClass("champion")).toEqual(true);

    RoundRowComponent.setProps({ seasonChampion: false });
    expect(RoundRowComponent.find(".root").hasClass("champion")).toEqual(false);
  });
});
