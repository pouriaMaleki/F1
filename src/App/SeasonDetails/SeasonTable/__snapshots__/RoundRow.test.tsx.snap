// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render RoundRow render correctly RoundRow component 1`] = `
<div
  className="root champion"
>
  <div
    className="data roundNo"
  >
    1
  </div>
  <div
    className="data"
  >
    Austali
  </div>
  <div
    className="data"
  >
    Lwoid
  </div>
  <div
    className="data"
  >
    McDAS
  </div>
  <div
    className="data"
  >
    12
     Laps
  </div>
  <div
    className="data"
  >
    2
     Grid
  </div>
  <div
    className="data"
  >
    1.24.32.433
  </div>
</div>
`;
